src/CollectorGuess.cpp

```diff
@@ -69,6 +69,12 @@ bool isDouble(const std::string& x, LocaleInfo* pLocale) {
   return parseDouble(pLocale->decimalMark_, begin, end, res) && begin == end;
 }
 
+bool isTime(const std::string& x, LocaleInfo* pLocale) {
+  DateTimeParser parser(pLocale);
+  parser.setDate(x.c_str());
+  return parser.parse(pLocale->timeFormat_);
```

Comment:
I think `parseTime()` is too flexible actually. Using it the following tests fail

``` r
collector_guess("0001")
#> [1] "time"
collector_guess("00010203")
#> [1] "time"
```